src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -209,8 +209,8 @@ public function getDefaultValue()
      */
     public function addChild(NodeInterface $node)
     {
-        $name = $node->getName();
-        if (!strlen($name)) {
+        $name = (string) $node->getName();
```

Comment:
Well, we don't do any checking for scalar types in Symfony. But the docblock clearly states that you have to pass string values. Thus, yes, please revert that change. But I don't think it is necessary to open an issue for that.